Stratis.Bitcoin.Features.BlockStore/LoopSteps/ProcessPendingStorageStep.cs

```diff
@@ -43,132 +42,158 @@ internal ProcessPendingStorageStep(BlockStoreLoop blockStoreLoop, ILoggerFactory
         /// <inheritdoc/>
         internal override async Task<StepResult> ExecuteAsync(ChainedBlock nextChainedBlock, CancellationToken cancellationToken, bool disposeMode)
         {
-            this.logger.LogTrace("({0}:'{1}/{2}',{3}:{4})", nameof(nextChainedBlock), nextChainedBlock.HashBlock, nextChainedBlock.Height, nameof(disposeMode), disposeMode);
+            this.logger.LogTrace("{0}:'{1}/{2}',{3}:{4}", nameof(nextChainedBlock), nextChainedBlock?.HashBlock, nextChainedBlock?.Height, nameof(disposeMode), disposeMode);
 
-            var context = new ProcessPendingStorageContext(this.BlockStoreLoop, nextChainedBlock);
+            var context = new ProcessPendingStorageContext(this.BlockStoreLoop, nextChainedBlock, cancellationToken);
 
             // Next block does not exist in pending storage, continue onto the download blocks step.
             if (!this.BlockStoreLoop.PendingStorage.ContainsKey(context.NextChainedBlock.HashBlock))
                 return StepResult.Next;
 
-            // If pending storage has not yet reached the threshold or the node is in IDB, stop the loop and wait for more blocks.
-            if (this.BlockStoreLoop.ChainState.IsInitialBlockDownload == true &&
-                disposeMode == false &&
-                this.BlockStoreLoop.PendingStorage.Count < BlockStoreLoop.PendingStorageBatchThreshold)
-                return StepResult.Stop;
+            if (disposeMode)
+                return await ProcessWhenDisposing(context);
+
+            if (this.BlockStoreLoop.ChainState.IsInitialBlockDownload)
+                return await ProcessWhenInIBD(context);
+
+            if (!this.BlockStoreLoop.ChainState.IsInitialBlockDownload)
+                return await ProcessWhenNotInIBD(context);
+
+            return StepResult.Stop;
```

Comment:
Seems we don't need this return 'StepResult.Stop;'

Reply:
Have fixed this with en "else"